com.unity.shadergraph/Editor/Generation/Processors/Generator.cs

```diff
@@ -703,27 +733,38 @@ void GenerateShaderPass(int targetIndex, PassDescriptor pass, ActiveFields activ
                     vertexGraphInputName,
                     vertexGraphFunctionName,
                     vertexGraphOutputName);
+                Profiler.EndSample();
 
                 // Generate final shader strings
-                vertexBuilder.AppendLines(vertexGraphOutputBuilder.ToString());
-                vertexBuilder.AppendNewLine();
-                vertexBuilder.AppendLines(vertexGraphFunctionBuilder.ToString());
+                if (m_humanReadable)
+                {
+                    vertexBuilder.AppendLines(vertexGraphOutputBuilder.ToString());
+                    vertexBuilder.AppendNewLine();
+                    vertexBuilder.AppendLines(vertexGraphFunctionBuilder.ToString());
+                }
+                else
+                {
+                    vertexBuilder.Append(vertexGraphOutputBuilder.ToString());
```

Comment:
FYI -- I think we can write a more optimized append here that gets rid of the ToString()-- just calls StringBuilder.Append(StringBuilder) internally, which will save some GC alloc and mem bandwidth.